optuna/storages/_base.py

```diff
@@ -558,6 +558,15 @@ def get_n_trials(self, study_id: int, state: Optional[TrialState] = None) -> int
         """
         raise NotImplementedError
 
+    @abc.abstractclassmethod
+    def get_url(self) -> Optional[str]:
+        """Get a path to a storage.
+
+        Returns:
+            A physical path to a storage.
+        """
+        raise NotImplementedError
```

Comment:
The design of this method seems reasonable because of the following reasons:
1. `storage` argument of `optuna.create_study` accepts `str`, and all storages can have the URL string.
1. Some storages may not have the URL string (i.e., in-memory storage for Dask integration discussed in https://github.com/optuna/optuna/issues/1766), but such storage can raise errors.
1. Preserving `None` for `InMemoryStorage` is consistent with the other part of Optuna.

`BaseStorage` may be able to have `url` property, but I prefer the current proposal because of the second reason. Raising errors in the property does not seem to be a good practice.

This change will affect the public interface of the storage, and I'd like to hear opinions of developers and other reviewers, so your comments are more than welcome. (cc: @keisuke-umezawa @hvy @ytsmiling)